br/pkg/task/stream.go

```diff
@@ -1753,74 +1752,65 @@ type PiTRTaskInfo struct {
 			return checkInfo, errors.Trace(err)
 		}
-		if exists {
-			curTaskInfo, err = checkpoint.LoadCheckpointTaskInfoForLogRestore(ctx, s, clusterID)
-			if err != nil {
-				return checkInfo, errors.Trace(err)
-			}
+		// the log restore checkpoint metadata is persist, so the PITR is in the log restore stage.
+		if curTaskInfo.Metadata != nil {
 			// TODO: check whether user has manually modified the cluster(ddl). If so, regard the behavior
 			//       as restore from scratch. (update `curTaskInfo.RewriteTs` to 0 as an uninitial value)
 
-			// The task info is written to external storage without status `InSnapshotRestore` only when
-			// id-maps is persist into external storage, so there is no need to do snapshot restore again.
-			if curTaskInfo.StartTS == cfg.StartTS && curTaskInfo.RestoreTS == cfg.RestoreTS {
-				// the same task, check whether skip snapshot restore
-				doFullRestore = doFullRestore && (curTaskInfo.Progress == checkpoint.InSnapshotRestore)
-				// update the snapshot restore task name to clean up in final
-				if !doFullRestore && (len(cfg.FullBackupStorage) > 0) {
-					_ = cfg.generateSnapshotRestoreTaskName(clusterID)
-				}
-				log.Info("the same task", zap.Bool("skip-snapshot-restore", !doFullRestore))
-			} else {
-				// not the same task, so overwrite the taskInfo with a new task
-				log.Info("not the same task, start to restore from scratch")
-				errTaskMsg = fmt.Sprintf(
-					"a new task [start-ts=%d] [restored-ts=%d] while the last task info: [start-ts=%d] [restored-ts=%d] [skip-snapshot-restore=%t]",
-					cfg.StartTS, cfg.RestoreTS, curTaskInfo.StartTS, curTaskInfo.RestoreTS, curTaskInfo.Progress == checkpoint.InLogRestoreAndIdMapPersist)
-
-				curTaskInfo = nil
+			if curTaskInfo.Metadata.UpstreamClusterID != cfg.upstreamClusterID {
+				return checkInfo, errors.Errorf(
+					"The upstream cluster id[%d] of the current log restore does not match that[%d] recorded in checkpoint. "+
+						"Perhaps you should specify the last log backup storage instead, "+
+						"or just clean the checkpoint database[%s] if the cluster has been cleaned up.",
+					cfg.upstreamClusterID, curTaskInfo.Metadata.UpstreamClusterID, checkpoint.LogRestoreCheckpointDatabaseName)
 			}
+
+			if curTaskInfo.Metadata.StartTS != cfg.StartTS || curTaskInfo.Metadata.RestoredTS != cfg.RestoreTS {
+				return checkInfo, errors.Errorf(
+					"The current log restore want to restore cluster from %d to %d, "+
+						"which is different from that from %d to %d recorded in checkpoint. "+
+						"Perhaps you should specify the last full backup storage to match the start-ts and "+
+						"the parameter --restored-ts to match the restored-ts. "+
+						"or just clean the checkpoint database[%s] if the cluster has been cleaned up.",
+					cfg.StartTS, cfg.RestoreTS, curTaskInfo.Metadata.StartTS, curTaskInfo.Metadata.RestoredTS, checkpoint.LogRestoreCheckpointDatabaseName,
+				)
+			}
+
+			log.Info("the same task, skip snapshot restore")
```

Comment:
It's better to have a clear log info.
```suggestion
			log.Info("detect log restore checkpoint. so skip snapshot restore and start log restore from the checkpoint")
```